### PR TITLE
Added dry-run option to prepare-release command

### PR DESCRIPTION
**Pull request by mitchellss**
Resolves: #128 

## How to Test

1. Run ```yactl new -m "<your message here>"```
2. Run the command ```yactl prepare-release --dry-run``` from the master branch of a git project
3. Verify that CHANGELOG.md is unchanged and that the change preview contains the earlier added change
4. Run ```yactl prepare-release --dry-run --quiet``` and verify a contradiction error is printed

## Potential Side Effects/Things to Look Out For

Kebab-case option differs from the rest of the camel-cased options which required a slight workaround in the validate-argv-middleware.

### CODE REVIEW

File: docs/COMMANDS.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ Gather the changelogs from `--logsDir` and compile them into `--changelogFile` u
...
 | `--validationPattern`    |       | A regular expression used to validate each individual changelog entry                                                                                                                                                                      | `string`    | `false`  | `\[.*\]\s+.*\s{#[\d]+}`                                                                                                                                                   |                    |
+| `--dry-run`              |       | Print out changes to be made without writing them.                                                                                                                                                                                         | `boolean`   | `false`  | `false`                                                                                                                                                                   |                    |
```
Comment (mrjones2014) [inline docs/COMMANDS.md:57]: I had forgotten the rest of the options were `camelCase` -- let's make this one follow suit for consistency.

**Review by mrjones2014 — CHANGES_REQUESTED** [review]
Been a while since I worked on this project, I forgot the rest of the options are `camelCase` -- not sure why I did it that way but let's make `--dryRun` camelCase so it's consistent with the others.

File: docs/COMMANDS.md (hunk 1 above)
Comment (mitchellss) [reply docs/COMMANDS.md]: Makes sense. It should be all fixed now.

**Review by mrjones2014 — APPROVED** [review]
looks good to me

**Pull request merged**